library/core/src/array/mod.rs

```diff
@@ -364,3 +364,61 @@ macro_rules! array_impl_default {
 }
 
 array_impl_default! {32, T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T}
+
+#[cfg(not(bootstrap))]
+#[lang = "array"]
+impl<T, const N: usize> [T; N] {
+    /// Returns an array of the same size as `self`, with function `f` applied to each element.
+    /// The closure will be called on elements 0 up to but excluding N.
```

Comment:
I think last sentence is kind of redundant. "applied to each element" should be sufficient, right?

Reply:
Someone suggested that I specify the order, as it's a `FnMut` closure, so it does matter how the function is called. Should I try condensing the sentences?

Reply:
Ah mh. I am not quite sure if we want to guarantee order. But I guess it's fine for now. Maybe just add "... in order" to the first sentence? Is that clear enough?

Reply:
(I'm the one who suggested documenting the order)

I suggested documenting the order because if it's `FnMut`, it's likely that people will start to depend on it, and when it's not documented there are often follow-up PRs requesting that it get documented anyway (for example, https://github.com/rust-lang/rust/pull/60396 for `retain`).

For APIs where we don't want to guarantee order than I'd personally encourage just `Fn` (though that's not an official libs guideline) because people shouldn't be surprised if they use interior mutability and the behaviour goes a bit weird.  But I think this is a case where different orders don't seem interesting, so I'd just pick "we go forward" and call it done.  (I would certainly expect that this would work like `.into_iter().map(f).collect()`.)

Reply:
Yeah, you are probably right; I can't imagine what not specifying the order would buy us. So yeah, `FnMut` and guaranteeing the obvious order seems fine! We should add it as open questions to the tracking issue though.

I would still improve this sentence as mentioned above.